### PR TITLE
extend wastime demo to support wasi32-wasi

### PR DESCRIPTION
**Pull request by stlankes**
To support wasi, the PR start to the "wasi_snapshot_preview1". Cuurently, only the basic functions are supported, which are needed to run the fibonacci example.

In addition, the fibonacci example is converted from a library to a common application.

### CODE REVIEW

File: examples/wasmtime/Cargo.toml [hunk 1]
```diff
@@ -12,9 +12,11 @@ anyhow = "1.0"
...
-simple_logger = { version = "5.0", default-features = false, features = ["nightly"]}
-wasmtime = { version = "22.0", default-features = false, features = ["std", "runtime", "cranelift", "threads", "gc", "component-model"] }
+simple_logger = { version = "5.0", default-features = false }
+wasmtime = { version = "22.0", default-features = false, features = ["std", "runtime", "cranelift", "threads"] } #, "gc", "component-model"] }
+zerocopy = { version = "0.7" }
 
 [target.'cfg(target_os = "hermit")'.dependencies]
 hermit = { path = "../../hermit", default-features = false, features = ["smp", "acpi", "pci", "fsgsbase", "fs", "tcp", "dhcpv4", "mmap"] }
 hermit-abi = { path = "../../hermit-abi", default-features = false }
+hermit-sync = { version = "0.1" }
```
Comment (mkroening) [inline examples/wasmtime/Cargo.toml:22]: Can't we use [`std::cell::OnceCell`](https://doc.rust-lang.org/stable/std/cell/struct.OnceCell.html) instead?

File: examples/wasmtime/src/preview1.rs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		.unwrap();
+	linker
+		.func_wrap(
+			"wasi_snapshot_preview1",
+			"args_sizes_get",
+			|mut caller: Caller<'_, _>, number_args_ptr: i32, args_size_ptr: i32| {
+				// Currently, we ignore the arguments
+				if let Some(Extern::Memory(mem)) = caller.get_export("memory") {
+					// Currently, we ignore the environment
+					let zereo: u32 = 0;
```
Comment (mkroening) [inline examples/wasmtime/src/preview1.rs:125]:
```suggestion
					let zero: u32 = 0;
```

File: examples/wasmtime/src/preview1.rs [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+		)
+		.unwrap();
+	linker
+		.func_wrap(
+			"wasi_snapshot_preview1",
+			"environ_sizes_get",
+			|mut caller: Caller<'_, _>, number_env_variables_ptr: i32, env_buffer_size_ptr: i32| {
+				if let Some(Extern::Memory(mem)) = caller.get_export("memory") {
+					// Currently, we ignore the environment
+					let zereo: u32 = 0;
```
Comment (mkroening) [inline examples/wasmtime/src/preview1.rs:159]:
```suggestion
					let zero: u32 = 0;
```

File: examples/wasmtime/src/main.rs [hunk 4]
```diff
@@ -53,79 +36,53 @@ fn main() -> Result<()> {
...
+	/*let imports = module.imports();
 	for i in imports {
 		info!("import from module `{}` symbol `{}`", i.module(), i.name());
 	}
 
+	let exports = module.exports();
+	for i in exports {
+		info!("exports symbol `{}`", i.name());
+	}*/
+
```
Comment (mkroening) [inline examples/wasmtime/src/main.rs:50]: Do we still need this?

```suggestion
```

**Review by mkroening — CHANGES_REQUESTED** [review]
This looks good to me overall! :)

**Comment by stlankes** [comment]
@mkroening Merge it. I fullfil all your requests.

**Review by mkroening — APPROVED** [review]

**Pull request merged**